src/Data/List.purs

```diff
@@ -239,9 +244,7 @@ head (Cons x _) = Just x
 -- |
 -- | Running time: `O(n)`.
 last :: forall a. List a -> Maybe a
-last (Cons x Nil) = Just x
-last (Cons _ xs)  = last xs
-last _            = Nothing
+last = head <<< reverse
```

Comment:
I'm pretty sure this will be a lot slower than the original definition - reconstructing the entire list and taking an element vs reading it once and taking the last element?